src/Zend/Session.php

```diff
@@ -694,7 +694,22 @@ public static function writeClose($readonly = true)
             parent::$_writable = false;
         }
 
+        /*
+         * In some cases (when php processes run at the same time) a race condition can occur
+         * This error handler catches specific warnings or throws an exception otherwise
+         */
+        set_error_handler(function ($num, $str, $file, $line, $context = null) {
+            if (strpos($str, 'Failed to write session data using user defined save handler') !== false) {
```

Comment:
I'm a little concerned about suppressing the existing error handling for this message in all cases, as I do wonder if, under different circumstances, logging this message could be valuable in debugging an underlying problem.

Would it be possible to make this a configurable behavior, perhaps a `suppress_session_write_warning` option or similar? I believe the options are set here:
https://github.com/diablomedia/zf1-session/blob/master/src/Zend/Session.php#L105-L116

I _think_ setting `parent::$_writable = false` does make sense regardless of that configuration option, though.

Reply:
Thanks for the feedback!

**exception**
You are absolutely right - throwing  an exception would change the behavior.  
With the new commit a user warning will be triggered instead of an exception. The output looks the same as if the there is no error_handler at all. 

**making it configurable**  
As you suggested I added an option/property for suppressing the warning. Per default the option is true and the warning message will be supressed.

**calling the previous error handler inside custom error handler**
Correct me if I'm wrong, but I think this will not work, because:

1. no error handler is set per default --> function will return NULL
1. previous error handler not accessable: the result of set_error_handler is only available after the definition of the custom error handler
1. possible infinte error handler loop

That's the reason why I changed the exception to a ``E_USER_WARNING``.